src/intelligence_layer/learning/postgres_instruction_finetuning_data_repository.py

```diff
@@ -62,18 +84,26 @@ def sample(self, id: str) -> Optional[InstructionFinetuningSample]:
             return db_sample.to_pydantic() if db_sample else None
 
     def samples(self, ids: Iterable[str]) -> Iterable[InstructionFinetuningSample]:
-        page, page_size = 1, 100
-        offset = (page - 1) * page_size
-        with self.Session.begin() as session:
-            db_samples = (
-                session.query(InstructionFinetuningSample_)
-                .filter(InstructionFinetuningSample_.id.in_(ids))
-                .offset(offset)
-                .limit(page_size)
-                .all()
-            )
-            for db_sample in db_samples:
-                yield db_sample.to_pydantic()
+        page = 1
+
+        while True:
+            offset = (page - 1) * self.page_size
+            with self.Session.begin() as session:
```

Comment:
I think we should move the session outside of the loops, s.t. it is not recreated in every iteration